integration-tests/src/test/java/tachyon/client/RemoteBlockInStreamIntegrationTest.java

```diff
@@ -535,4 +541,42 @@ public void seekAroundLocalBlock() throws IOException, TachyonException {
     Assert.assertEquals(99, is.read());
     is.close();
   }
+
+  /**
+   * Test remote read stream lock in <code>RemoteBlockInStream</code>.
+   */
+  @Test
+  public void remoteReadLockTest() throws Exception {
+    String uniqPath = PathUtils.uniqPath();
+    for (int k = MIN_LEN + DELTA; k <= MAX_LEN; k += DELTA) {
+      TachyonFile f =
+          TachyonFSTestUtils.createByteFile(mTfs, uniqPath + "/file_" + k, k, mWriteTachyon);
+
+      long blockId = mTfs.getInfo(f).getBlockIds().get(0);
+      BlockInfo info = TachyonBlockStore.get().getInfo(blockId);
+
+      NetAddress workerAddr = info.getLocations().get(0).getWorkerAddress();
+      RemoteBlockInStream is =
+          new RemoteBlockInStream(info.getBlockId(), info.getLength(),
+              new InetSocketAddress(workerAddr.getHost(), workerAddr.getDataPort()));
+      byte[] ret = new byte[k / 2];
+      Assert.assertEquals(0, is.read());
+      mTfs.delete(f);
+      CommonUtils.sleepMs(mWorkerToMasterHeartbeatIntervalMs);
+
+      checkFileDeleted(f);
+      Assert.assertEquals(k / 2, is.read(ret, 0, k / 2));
```

Comment:
Add comments to this test explaining what it is testing, e.g. `// Look! We can still read the deleted file since we have a lock!`